test/projects/diff/highlight.html

```diff
@@ -0,0 +1,270 @@
+          linesChanged: { startLineBefore: 340, rangeBefore: 6,
+                          startLineAfter: 338,  rangeAfter: 8 }
+        },
+        {
+          file: '14',
+          diffId: '@@ -343,3 +341,5 @@',
+          fileUrl: 'a',
+          changes: "\n Pellentesque a nulla a libero molestie blandit vitae id eros.\n" +
+                   " Morbi fermentum neque sit amet ante eleifend, non molestie nulla pretium.\n" +
+                   " Integer non turpis eu quam bibendum pulvinar vel non ante.\n" +
+                   "+Etiam vel nibh aliquam, dignissim nunc vel, ultrices sapien.\n" +
+                   "+Nulla ac justo non tellus convallis suscipit.\n",
+          linesChanged: { startLineBefore: 343, rangeBefore: 3,
+                          startLineAfter: 341,  rangeAfter: 5 }
+        },
+        {
+          file: '14',
+          diffId: '@@ -341,5 +343,3 @@',
+          fileUrl: 'a',
+          changes: "\n Pellentesque a nulla a libero molestie blandit vitae id eros.\n" +
+                   " Morbi fermentum neque sit amet ante eleifend, non molestie nulla pretium.\n" +
+                   " Integer non turpis eu quam bibendum pulvinar vel non ante.\n" +
+                   "-Etiam vel nibh aliquam, dignissim nunc vel, ultrices sapien.\n" +
+                   "-Nulla ac justo non tellus convallis suscipit.\n",
+          linesChanged: { startLineBefore: 341, rangeBefore: 5,
+                          startLineAfter: 343,  rangeAfter: 3 }
+        }
+      ]
+      var diffElem;
+
+      setup(function() {
+        diffElem = fixture('basic');
+      });
+
+      test('renders when attached', function(done) {
+        var elem = document.createElement('pull-request-diff-highlight');
+        elem._render = function() {
+          done();
+        }
+        document.body.appendChild(elem);
+      });
+
+      test('renders when code changes', function(done) {
+        diffElem._render = function() {
+          done();
+        }
+        diffElem.code = "+ insert!";
+      });
+
+      testDiffs.forEach(function(diff) {
+        test('shows correct amount of `before` lines for diff ' + diff.file, function() {
+          diffElem.code = diff.changes;
+          diffElem.linesChanged = diff.linesChanged;
+
+          var beforeLines = filterEmpty(getChildrenForId('beforeColumn'), '');
+          assert.equal(diff.linesChanged.rangeBefore, beforeLines.length);
+        });
+
+        test('shows correct amount of `after` lines for diff ' + diff.file, function() {
+          diffElem.code = diff.changes;
+          diffElem.linesChanged = diff.linesChanged;
+
+          var afterLines = filterEmpty(getChildrenForId('afterColumn'), '');
+          assert.equal(diff.linesChanged.rangeAfter, afterLines.length);
+        });
+
+        test('shows correct amount of empty `before` lines for diff ' + diff.file, function() {
+          diffElem.code = diff.changes;
+          diffElem.linesChanged = diff.linesChanged;
```

Comment:
Hm foreach, not sure how that works in tests. Could you maybe extract it to a `suite` with a `setUp` where you set the code and linesChanged?